### ISSUE TITLE
Add Frontmatter option to exclude page from flexsearch indexing

### ISSUE THREAD

**Issue by evoteal** [gethinode/hinode#636]
**Is your feature request related to a problem? Please describe.**
FlexSearch currently indexes sites based on the presence of a title. Excluding a site requires removing its title, which breaks the breadcrumb navigation.

**Describe the solution you'd like**
Introduce a "searchExclude" frontmatter option to prevent specific sites from appearing in search results.

**Describe alternatives you've considered**
Open to suggestions for other methods to exclude sites from search results.

**Additional context**
 Need to exclude large legal texts like privacy policies and imprints, which are frequently indexed but should not be displayed at all.

**Comment by markdumay** [comment]
The feature has been added to [v0.21.8](https://github.com/gethinode/hinode/releases/tag/v0.21.8). You can set `searchExclude = false` in the frontmatter to exclude an individual page from the search index. You may need to restart your hugo server and refresh your browser to ensure the JavaScript file is loaded correctly.

**Comment by evoteal** [comment]
Wow, thank you for your quick implementation. However, I've come across a bug related to the file structure which includes: content/a.md, content/b.md, and content/c.md. I can exclude a.md and b.md from search results using "searchExclude: true", but I cannot do the same for c.md. The issue seems to hinge on the alphabetical ordering of files and how the flexsearch module uses .add to index pages. Specifically, if the last page in alphabetical order has "searchExclude: true" set, it results in incorrect JavaScript code generation. This issue affects the search functionality across the entire site, including subfolders like content/posts. I was able to reproduce this bug using different filenames and after each change, I restarted the server and cleared the browser cache/reloaded the page.
